synapseclient/core/multithread_download/download_threads.py

```diff
@@ -340,8 +340,7 @@ def session_get():
             # but for now matching previous behavior.
             response = with_retry(
                 session_get,
-                retry_status_code_in=False,
-                retry_status_codes=(HTTPStatus.PARTIAL_CONTENT,),
+                expected_status_codes=(HTTPStatus.PARTIAL_CONTENT,),
```

Comment:
just double checking, we still want to retry on the usual `[429, 500, 502, 503, 504]` right?

Reply:
With the expected_status_codes passed it short circuits the retry_status_codes, and in this case with it will retry on anything other than a 206 PARTIAL CONTENT which seems a bit strange, but that was the previous behavior (except we would do it rapidly with no backoff).